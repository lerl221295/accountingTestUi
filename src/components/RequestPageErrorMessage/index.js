import React from 'react';
import PropTypes from 'prop-types';

const RequestPageErrorMessage = ({ message, onlyMessage }) => (
  <div className="page-error-message">
    {
      !onlyMessage
        ? (<div className="error-page">
          <div className="title">
400
          </div>
          <h4 className="message">
{message}
          </h4>
           </div>)
      : (<h2>
{message}
         </h2>)
    }
  </div>
);

RequestPageErrorMessage.defaultProps = {
  onlyMessage: false
};

RequestPageErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
  onlyMessage: PropTypes.bool
};

export default RequestPageErrorMessage;
