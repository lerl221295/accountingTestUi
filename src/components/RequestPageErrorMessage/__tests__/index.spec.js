
/* eslint no-undef: 0 */
import React from 'react';
// import renderer from 'react-test-renderer';

import RequestPageErrorMessage from '../index';

describe.skip('LoginErrorMessage Component', () => {

    test('Fail on ommit required message prop', () => {
        const spy = jest.spyOn(console, 'error');

        const SUT = renderer.create(
            <RequestPageErrorMessage />
        );

        expect(spy).toHaveBeenCalled();

        if (Array.isArray(spy.mock.calls) && spy.mock.calls.length > 0) {
            const consoleCall = spy.mock.calls[0][0];
            const propTypeRequired = consoleCall.split('.')[0];

            expect(propTypeRequired)
            .toBe('Warning: Failed prop type: The prop `message` is marked as required in `RequestLoginErrorMessage`, but its value is `undefined`');
        } else {
            throw Error('No estÃ¡ controlado si `message` es requerido!');
        }

        spy.mockReset();
        spy.mockRestore();
    });


    test('Render OK with message', () => {
        const props = {
            message: 'Mensaje de error para testing.'
        };

        const component = renderer.create(
            <RequestPageErrorMessage message={props.message} />
        );

        const SUT = component.toJSON();

        expect(SUT).toMatchSnapshot();
    });


    test('Render OK with onlyMessage enabled', () => {
        const props = {
            message: 'Mensaje de error para testing.'
        };

        const component = renderer.create(
        <RequestPageErrorMessage
          message={props.message}
          onlyMessage
        />
        );

        const SUT = component.toJSON();

        expect(SUT).toMatchSnapshot();
    });

});
