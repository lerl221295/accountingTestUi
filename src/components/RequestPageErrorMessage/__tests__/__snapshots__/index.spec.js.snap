// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginErrorMessage Component Render OK with message 1`] = `
<div
  className="login-error-message"
>
  <div
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      }
    }
  >
    <div
      className="error-page"
    >
      <div
        className="title"
      >
        5
        <span>
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fill": "currentColor",
                "height": 45,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "verticalAlign": "bottom",
                "width": 45,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
            />
          </svg>
        </span>
        0
      </div>
      <h4
        className="message"
      >
        Mensaje de error para testing.
      </h4>
    </div>
  </div>
</div>
`;

exports[`LoginErrorMessage Component Render OK with onlyMessage enabled 1`] = `
<div
  className="login-error-message"
>
  <h2>
    Mensaje de error para testing.
  </h2>
</div>
`;
